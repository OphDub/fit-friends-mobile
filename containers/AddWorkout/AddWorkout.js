import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableHighlight,
  FlatList,
} from 'react-native';

export default class AddWorkout extends Component {
  constructor(props) {
    super(props);
    this.state = {
      workoutName: '',
      workoutDesc: '',
      exerciseName: '',
      reps: '',
      exercises: [],
      error: ''
    };
  }

  addExercise = () => {
    const { reps, exerciseName } = this.state;

    if(!this.exerciseValidation()) {
      return;
    }

    const id = Date.now();
    const newExercise = Object.assign({ id, reps, name: exerciseName });
    const exercises = [...this.state.exercises, newExercise];

    this.setState({
      exercises,
      reps: '',
      exerciseName: ''
    });
  }

  exerciseValidation = () => {
    const { reps, exerciseName } = this.state;

    if (reps === '' || exerciseName === '') {
      const error = 'Please give your exercise a rep count and name.';

      this.setState({ error });
      return false;
    }

    return true;
  }

  removeExercise = (id) => {
    const { exercises } = this.state;
    const filteredExercises = exercises.filter(exercise => exercise.id !== id);

    this.setState({
      exercises: filteredExercises
    });
  }

  renderExercises = () => {
    return (
      <FlatList
        data={this.state.exercises}
        renderItem={ ({exercise}) =>
          <TouchableHighlight>
            <Text>{exercise.name}</Text>
            <Text>{exercise.reps}</Text>
          </TouchableHighlight>
        }
      />
    )
  }

  render() {
    return(
      <View style={styles.container}>
        <Text>Add Workout</Text>
        <Text>Workout Name: </Text>
          <TextInput
            style={styles.inputs}
            value={this.state.workoutName}
            keyboardAppearance={'dark'}
            onChangeText={(workoutName) => this.setState({ workoutName })}
          />
        <Text>Description (Optional):</Text>
          <TextInput
            style={styles.inputs}
            value={this.state.workoutDesc}
            keyboardAppearance={'dark'}
            onChangeText={(workoutDesc) => this.setState({ workoutDesc })}
          />
        <Text>Exercises:</Text>
          <Text>Exercise Name:</Text>
          <TextInput
            style={styles.inputs}
            value={this.state.exerciseName}
            keyboardAppearance={'dark'}
            onChangeText={(exerciseName) => this.setState({ exerciseName })}
          />
          <Text>Reps:</Text>
          <TextInput
            style={styles.inputs}
            value={this.state.reps}
            keyboardAppearance={'dark'}
            keyboardType={'number-pad'}
            onChangeText={(reps) => this.setState({ reps })}
          />
          <TouchableHighlight
            style={styles.buttons}
            onPress={this.addExercise}
          >
            <Text>Add Exercise</Text>
          </TouchableHighlight>
          <TouchableHighlight
            style={styles.buttons}
            onPress={this.addWorkout}
          >
            <Text>Add Workout</Text>
          </TouchableHighlight>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#D3D3D3',
    marginVertical: 20,
    marginLeft: 5,
    marginRight: 5,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 10,
  },
  inputs: {
    backgroundColor: '#ffff',
    borderRadius: 5,
    width: 120,
  },
  buttons: {
    backgroundColor: '#ffff',
    padding: 10,
  }
});