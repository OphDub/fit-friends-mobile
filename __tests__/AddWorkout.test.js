import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddWorkout from '../containers/AddWorkout/AddWorkout';

describe('AddWorkout Component', () => {
  let rendered;

  beforeEach(() => {
    rendered = shallow(<AddWorkout/>);
  });

  it('should match the snapshot', () => {
    rendered = renderer.create(<AddWorkout />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('workoutName and workoutDesc in state should start as empty strings', () =>{
    const expected = '';

    expect(rendered.instance().state.workoutName).toEqual(expected);
    expect(rendered.instance().state.workoutDesc).toEqual(expected);
  });

  it('exercises array in state should should be empty', () => {
    const expected = [];

    expected(rendered.instance().state.exercises).toDeepEqual(expected);
  });

  it('handleInputChange should update state', () => {

  });
});
