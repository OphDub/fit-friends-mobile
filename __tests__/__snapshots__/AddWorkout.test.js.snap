// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWorkout Component should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Add Workout
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Workout Name: 
  </Text>
  <TextInput
    allowFontScaling={true}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Description (Optional):
  </Text>
  <TextInput
    allowFontScaling={true}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Exercises
  </Text>
  <TextInput
    allowFontScaling={true}
  />
</View>
`;
